include/fmt/iterator.h

```diff
@@ -0,0 +1,81 @@
+// Copyright (c) Google LLC.
+// SPDX-License-Identifier: MIT Licence
+//
+
+#ifndef FMT_ITERATOR_H_
+#define FMT_ITERATOR_H_
+
+#include <cstdio>
+
+#include "core.h"
+#include "format.h"
+
+FMT_BEGIN_NAMESPACE
+
+namespace internal {
+#if defined(__cpp_concepts) and __cpp_concepts >= 201907
+template <typename T> concept formattable = requires(T&& t) {
+  ::fmt::format("{}", std::forward<T>(t));
+};
+#endif // __cpp_concepts
+}  // namespace internal
+
+// Iterator that allows for writing to file via fmt::print.
+//
+template <typename CharT> class basic_print_iterator {
+ public:
+  using iterator_category = std::output_iterator_tag;
+  using value_type = void;
+  using difference_type = std::ptrdiff_t;
```

Comment:
C++20 changes this because you need an `iter_difference_t` to model `weakly_incrementable`.

See [`output_iterator`](http://eel.is/c++draft/ostream.iterator).

Reply:
Good to know. Then ignore my comment.